src/test/java/net/snowflake/client/jdbc/ResultSetAsyncIT.java

```diff
@@ -133,17 +130,24 @@ public void testResultSetMetadata() throws SQLException {
   }
 
   @Test
-  public void testisClosedIsLastisAfterLast() throws SQLException {
+  public void testOrderAndClosureFunctions() throws SQLException {
     // Set up environment
     Connection connection = getConnection();
-    final Map<String, String> params = getConnectionParameters();
     Statement statement = connection.createStatement();
     statement.execute("create or replace table test_rsmd(colA number(20, 5), colB string)");
     statement.execute("insert into test_rsmd values(1.00, 'str'),(2.00, 'str2')");
     ResultSet resultSet =
         statement.unwrap(SnowflakeStatement.class).executeAsyncQuery("select * from test_rsmd");
-    String queryID = resultSet.unwrap(SnowflakeResultSet.class).getQueryID();
+
+    // test isFirst, isBeforeFirst
+    assertTrue("should be before the first", resultSet.isBeforeFirst());
```

Comment:
Now added testing for isFirst, isBeforeFirst to the pre-existing test for isLast, isAfterLast